src/plugins/global-settings/preset-controls/index.php

```diff
@@ -0,0 +1,172 @@
+<?php
+/**
+ * Size and Spacing Preset Controls
+ */
+
+// Exit if accessed directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+if ( ! class_exists( 'Stackable_Size_And_Spacing_Preset_Controls' ) ) {
+
+	/**
+	 * Size and Spacing Preset Controls
+	 */
+    class Stackable_Size_And_Spacing_Preset_Controls {
+		public const PRESET_MAPPING = array(
+			'fontSizes' => array(
+				'settings' => array('typography', 'fontSizes' ),
+				'prefix' => 'font-size',
+			),
+			'spacingSizes' => array(
+				'settings' => array( 'spacing', 'spacingSizes' ),
+				'prefix' => 'spacing',
+			),
+			'blockHeights' => array(
+				'settings' => array( 'blockHeights' ),
+				'prefix' => 'block-height',
+			),
+			'borderRadius' => array(
+				'settings' => array( 'borderRadius' ),
+				'prefix' => 'border-radius',
+			),
+		);
+
+		public $custom_presets;
+		public $theme_presets;
+		public $default_presets;
+		public $stackable_presets;
+
+		/**
+		 * Initialize
+		 */
+  		function __construct() {
+			$this->custom_presets = get_option( 'stackable_global_custom_preset_controls' );
+			$this->theme_presets = WP_Theme_JSON_Resolver::get_theme_data()->get_settings();
+			$this->default_presets = WP_Theme_JSON_Resolver::get_core_data()->get_settings();
+			$this->stackable_presets = $this->load_presets( __DIR__ . '/presets.json');
+
+			add_filter( 'stackable_inline_styles_nodep', array( $this, 'add_preset_controls_styles' ) );
+			add_filter( 'stackable_inline_editor_styles', array( $this, 'add_preset_controls_styles' ) );
+		}
+
+		public static function sanitize_array_setting( $input ) {
+			return ! is_array( $input ) ? array( array() ) : $input;
+		}
+
+		private function load_presets( $json_path ) {
```

Comment:
Maybe change this function name since it's not actually loading presets, it's just loading the provided json file